### PR TITLE
Format code and fix spelling errors

### PR DESCRIPTION
**Pull request by jooola**
To be the one who will annoy on each commit, I propose to add a pre-commit config to format the code and check for spelling errors. This follows #1654.

There is also a CI job that enforce that.

The PR is quiet big, hope you don't mind.

### CODE REVIEW

File: .github/scripts/build-deb.sh [hunk 1]
```diff
@@ -1,4 +1,4 @@
-dch --create --distribution unstable --package "${LIQ_PACKAGE}" --newversion "1:${LIQ_VERSION}-${LIQ_TAG}-${DEB_RELEASE}" "Build ${COMMIT_SHORT}"#!/bin/sh
+dch --create --distribution unstable --package "${LIQ_PACKAGE}" --newversion "1:${LIQ_VERSION}-${LIQ_TAG}-${DEB_RELEASE}" "Build ${COMMIT_SHORT}" #!/bin/sh
```
Comment (jooola) [inline .github/scripts/build-deb.sh:1]: This is weird.

File: CHANGES.md [hunk 2]
```diff
@@ -1003,17 +994,17 @@ New:
 
 Enhancements:
 
-- Handle more OSC types (float, float_pair, bool, string, string_pair) and added
-  osc.on_*.
+- Handle more OSC types (float, float*pair, bool, string, string_pair) and added
+  osc.on*\*.
```
Comment (jooola) [inline CHANGES.md:998]: Formatting error

**Comment by toots** [comment]
This is awesome. Super down to integrate this. Having standardized format helps backport fixes over to rolling releases a lot.

Let us know when this is ready for in depth review!

**Comment by smimram** [comment]
> One thing is bothering me, how could we handle this spelling error:
> 
> https://github.com/savonet/liquidsoap/blob/ae732a784695fae92d7bac1d165b24b4e39c610f/src/core/io/alsa_io.ml#L92
> 
> I guess we have to send this upstream ? Or is this something we can change ?

We are upstream for `ocaml-alsa` :)

However, this is not a typo: it should be read as "write int" (with int shortened to "n"), and [this is the name already taken by the library](https://www.alsa-project.org/alsa-doc/alsa-lib/group___p_c_m.html#gae599772ce3d0aa6a70de143abcf145e7)

**Comment by toots** [comment]
> I think before marking this as ready for review, I would like to let the CI run to check everything is working as expected.

That's reasonable!

**Comment by jooola** [comment]
> > I think before marking this as ready for review, I would like to let the CI run to check everything is working as expected.
>
> That's reasonable!

I was hoping that you could allow the pipeline to run, because I am a first contributor here, and it prevent any pipeline runs.

**Comment by smimram** [comment]
> > > I think before marking this as ready for review, I would like to let the CI run to check everything is working as expected.
> > 
> > 
> > That's reasonable!
> 
> I was hoping that you could allow the pipeline to run, because I am a first contributor here, and it prevent any pipeline runs.

Launched.

**Comment by jooola** [comment]
Done

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -16,6 +16,38 @@ jobs:
...
+  pre-commit:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        with:
+          python-version: "3.x"
+      - uses: pre-commit/action@v3.0.0
+
+  check-shell:
```
Comment (toots) [inline .github/workflows/ci.yml:28]:
```suggestion
  check-spell:
```
😅

File: .pre-commit-config.yaml [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        args:
+          - --ignore-words=.codespellignore
+          - --builtin=clear,rare,informal
+
+  - repo: local
+    hooks:
+      - id: ocamlformat
+        name: ocamlformat
+        description: Formatting ML files
+        entry: ocamlformat
```
Comment (toots) [inline .pre-commit-config.yaml:42]: The location of this binary depends on opam env variables. Usually, you run `eval $(opam env)` first to set the proper env variables.

**Comment by jooola** [comment]
I'll work on another branch on my repo so I don't bother you with the workflow approval thing.

**Comment by jooola** [comment]
I will fix the shell scripts in another PR, the main branch moves way too fast to keep the branch up to date.

**Pull request closed without merging**